src/sparsezoo/model/utils.py

```diff
@@ -54,6 +57,49 @@
 SAVE_DIR = os.getenv("SPARSEZOO_MODELS_PATH", CACHE_DIR)
 
 
+class ModelResult(BaseModel):
+    """
+    Base class to store common result information
+
+    :param result_type: A string representing the typr of result
+        ex `training`, `inference`, etc
+    :param recorded_value: The float value of the result
+    :param recorded_units: The unit in which result is specified
+    """
+
+    result_type: str
+    recorded_value: float
+    recorded_units: str
+
+
+class ValidationResult(ModelResult):
+    """
+    A class holding information for validation results
+
+    :param dataset_type: A string representing the type of dataset used
```

Comment:
Do we care if it's downstream or upstream as long as the name of the dataset is provided?

Reply:
Yes that info is needed for `sparsify.package`